qt/place_panel.cpp

```diff
@@ -58,12 +57,66 @@ class QHLine : public QFrame
   }
 };
 
-PlacePageDialogUser::PlacePageDialogUser(QWidget * parent, place_page::Info const & info,
-                                         search::ReverseGeocoder::Address const & address)
-  : QDialog(parent)
+void PlacePanel::addCommonButtons(QDialogButtonBox * dbb, place_page::Info const & info)
+{
+  dbb->setCenterButtons(true);
+
+  QPushButton * fromButton = new QPushButton("From");
+  fromButton->setIcon(QIcon(":/navig64/point-start.png"));
+  fromButton->setAutoDefault(false);
+  connect(fromButton, &QAbstractButton::clicked, this, [this, &info](){ routeFrom(info); });
```

Comment:
@vng do you think this could lead to undefined behavior? Because we're capturing `info` by reference to this lambda that will run at a later point, and I'm not sure if `info` is deleted after `setPlace` is called. Is capturing `info` by copy an option (i.e., can `place_page::Info` be copied)?

Reply:
It should be fine here.
```Framework::GetCurrentPlacePageInfo()``` returns persistent object.
But the state can change when invoking lambda.

Reply:
Hmm then it's probably better to capture by copy.